### ISSUE TITLE
Storage Account creation sends API requests to the storage URL

### ISSUE THREAD

**Issue by adammontlake** [hashicorp/terraform-provider-azurerm#20625]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v1.3.6

### AzureRM Provider Version

3.30.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account,azurerm_key_vault

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "example-sa" {
  name                     = "name"
  resource_group_name      = azurerm_resource_group.rg.name
  location                 = azurerm_resource_group.rg.location
  account_tier             = local.account_tier
  account_replication_type = local.account_replication_type
}
```

### Debug Output/Panic Output

```shell
https://gist.github.com/adammontlake/c61e80882be3f381b453348f28728669
```

### Expected Behaviour

During the creation of a new resource, the only URL Terraform would reach out to is: management.azure.com

### Actual Behaviour

During the creation/update/destroy of a storage account, Terraform reaches out to three url's:

1. management.azure.com
2. https://StorageName.blob.core.windows.net/?comp=properties&restype=service
3. https://StorageName.queue.core.windows.net/?comp=properties&restype=service

The first URL is expected and creates the resource, the second and third are an issue because we are in a controlled environment and want to access the storage only through private networking (private endpoint) but the private endpoint creation happens only after the resource creation so during deployment these addresses resolve to a public domain which is not reachable, and causes Terraform to fail.

This also happens if "public_network_access_enabled" is disabled, and also if no blob or queue containers are created.
The same calls happen during update and delete operations.

Same issue exists for Key Vault.

### Steps to Reproduce

Deploy any Storage Account or Key Vault to Azure will cause these requests to be issued.

**Comment by tombuildsstuff** [comment]
hi @adammontlake 

Taking a look through here this appears to be a duplicate of https://github.com/hashicorp/terraform-provider-azurerm/issues/2977, which is blocked on the Service Team - rather than having multiple issues open tracking the same thing I'm going to close this issue in favour of that one, would you mind subscribing to that issue for updates?

Thanks!
